nad_ch/infrastructure/task_queue.py

```diff
@@ -0,0 +1,49 @@
+import os
+from celery import Celery
+
+
+class RedisTaskQueue:
+    def __init__(self, name: str, password: str, host: str, port: str):
+        redis_url = f"redis://:{password}@{host}:{port}/0"
+
+        self.app = Celery(name, broker=redis_url, backend=redis_url)
+
+        self.app.conf.update(
+            task_serializer="json", result_serializer="json", accept_content=["json"]
+        )
+
+    def define_task(self, func):
+        return self.app.task(func)
+
+    def start_worker(self):
+        self.app.worker_main(argv=["worker", "--loglevel=info"])
+
+
+class LocalTaskQueue:
```

Comment:
This looks great! I could also see maybe wanting an in-memory broker for unit tests.